### ISSUE TITLE
Volume Mixer

### ISSUE THREAD

**Issue by Solo187** [schellingb/dosbox-pure#59]
I can't seem to find any kind of volume mixer anywhere. Certain games have the music to low & the voices too high etc. Simon the sorcerer for one. & Chewy esc from F5. Probably quite a few others. Im sure the original Dosbox has something for this?

**Comment by schellingb** [comment]
That seems like a frontend thing and not a core thing? There's Audio -> Volume Gain in RetroArch.
Have you seen other libretro cores that have a volume mixer? If it's a common thing then I guess it can be implemented.

**Comment by Solo187** [comment]
I think it's a soundblaster mixer setting perhaps. There's commands in dosbox like mixer sb. Or mixer master. For volumes. Adjusting anywhere else will just turn everything up or down. Scummvm has a volume mixer where you can turn up or down sfx & music seperatley. If you try the games i mentioned you will see what i mean.

**Comment by schellingb** [comment]
The [MIXER commandline program](https://www.dosbox.com/wiki/MIXER) is available in DOSBox Pure as well. With it, you can control volume per channel.
It would be possible to expose the 10 volume settings (Master, Soundblaster, PC Speaker, Adlib, Disney, CD Audio, CMS, GUS, Tandy, Midi synth) to the core options but that seems overkill. I guess there could be an option for showing/hiding the volume options?

**Comment by Solo187** [comment]
Yes that's it im sure. Sounds good if you can implement that. I know in the puae core it has a ton of options like that which you can choose to show or hide

**Comment by eadmaster** [comment]
> It would be possible to expose the 10 volume settings (Master, Soundblaster, PC Speaker, Adlib, Disney, CD Audio, CMS, GUS, Tandy, Midi synth) to the core options but that seems overkill. I guess there could be an option for showing/hiding the volume options?

If implemented as core options, you could store per-game volume settings like i'm doing [here](https://github.com/eadmaster/libretro_volume_fixes).
As an alternative i guess it should be possible to store the same settings in a batch file and place it inside each game archive.

**Comment by Xenphor** [comment]
I agree that being able to have all the volume adjustments exposed to the core options would be very helpful. In many games I find either the FM synth or MIDI to be very low at default settings and the sound effects too overpowering. Most games don't feature in game volume sliders so it's hard to get a good balance. Thanks.

**Comment by fidoboy** [comment]
Is there any progress on this issue? I can't see any easy way to setup volume mixer per game. When using MT32 together with SB output for soundFX/voices the music is so often much louder than voices. ScummVM solves this easily with mixer settings into the interface, so if dos box pure provide it to RetroArch it can be also configured.
